src/contentScript/loadActivationEnhancements.ts

```diff
@@ -79,23 +75,24 @@ export async function loadActivationEnhancements(): Promise<void> {
     return;
   }
 
-  const installedRecipeIds = await getInstalledRecipeIds();
+  const activatedModIds = await getActivatedModIds();
 
   for (const button of activateButtonLinks) {
     const url = new URL(button.href);
-    let recipeId: RegistryId;
-    try {
-      recipeId = validateRegistryId(url.searchParams.get("id"));
-    } catch {
+
+    const modIds = extractIdsFromUrl(url.searchParams);
+
+    if (modIds.length === 0) {
       continue;
     }
 
-    // Check if recipe is already activated, and change button content to indicate active status
-    // Note: This should only run on the Marketplace page
-    if (isMarketplacePage() && installedRecipeIds.has(recipeId)) {
+    // Check if all mods are already activated, and change button content to indicate active status
```

Comment:
How does this sound?
```suggestion
    // Check if all mods in the pack are already activated, and change button content to indicate active status
```